docs/en/observability/profiling-upgrade.asciidoc

```diff
@@ -54,10 +56,359 @@ Integrations Server. To do this:
+  curl -s -f -o /dev/null -X POST -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/$old_index_name/_clone/.$alias_name-v001-000001" -H 'Content-Type: application/json' -d"
+  {
+    \"settings\": {
+      \"index.hidden\": true,
+      \"index.lifecycle.name\": \"profiling-60-days\",
+      \"index.lifecycle.rollover_alias\": \"$alias_name\",
+      \"index.blocks.write\": false,
+      \"index.lifecycle.origination_date\": $creation_date,
+      \"routing.allocation.include._tier_preference\": \"data_hot\"
+    },
+    \"aliases\": {
+      \"$alias_name\": {
+        \"is_write_index\": true
+      }
+    }
+  }"
+
+  curl -s -f -o /dev/null -X DELETE -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/$old_index_name"
+}
+
+function rename_regular_index {
+  local index_name=$1
+  curl -s -f -o /dev/null -X PUT -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/$index_name/_settings" -H 'Content-Type: application/json' -d'
+  {
+    "settings": {
+      "index.blocks.write": true
+    }
+  }'
+  local creation_date=$(curl -s -f -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/$index_name/_settings" | jq -r "..|.creation_date?|values")
+
+  curl -s -f -o /dev/null -X POST -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/$index_name/_clone/.$index_name-v001" -H 'Content-Type: application/json' -d"
+  {
+    \"settings\": {
+      \"index.hidden\": true,
+      \"index.blocks.write\": false,
+      \"index.lifecycle.origination_date\": $creation_date
+    }
+  }"
+
+  curl -s -f -o /dev/null -X DELETE -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/$index_name"
+  # Create the alias in a separate call after we've deleted the index
+  curl -s -f -o /dev/null -X POST -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/.$index_name-v001/_alias/$index_name" -H 'Content-Type: application/json' -d'
+  {
+    "is_write_index": true
+  }'
+}
+
+
+function migrate {
+  {
+    echo "Indices and aliases before migration"
+    curl -s -f -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/_cat/indices/*profiling*?v&s=index&expand_wildcards=all"
+    curl -s -f -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/_cat/aliases/*profiling*?v&expand_wildcards=all"
+  }  > "elastic_universal_profiling_migration_log.txt"
+
+  echo "Allowing wildcard deletes ..."
+  curl -s -f -o /dev/null -X PUT -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/_cluster/settings" -H 'Content-Type: application/json' -d'
+  {
+    "persistent": {
+      "action.destructive_requires_name": false
+    }
+  }'
+
+  echo "Disabling ILM temporarily..."
+  curl -s -f -o /dev/null -X POST -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/_ilm/stop"
+
+  echo "Deleting obsolete indices..."
+  curl -s -f -o /dev/null -X DELETE -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/.profiling-ilm-lock"
+  curl -s -f -o /dev/null -X DELETE -k -u "$ES_USER":"$ES_PASSWORD" "$ES_HOST/profiling-returnpads-private,profiling-sq-executables,profiling-sq-leafframes,profiling-symbols?ignore_unavailable=true"
+  # We rely only on the "next" indices

```

Comment:
As an explicit note: In the following lines we delete the "current" indices. We can't keep both easily (a custom reindex procedure would be possible but IMHO way too complex) so we only use the "next" indices which will contain data since the last rollover but not before. So customers may see partial data for older profiles.